pkgs/development/python-modules/twisted/default.nix

```diff
@@ -1,54 +1,153 @@
+    echo 'UNIXTestsBuilder_EPollReactorTests.test_sendFileDescriptorTriggersPauseProducing.skip = "sendFileDescriptor producer was not paused"'>> src/twisted/internet/test/test_unix.py
+    echo 'UNIXTestsBuilder_PollReactorTests.test_sendFileDescriptorTriggersPauseProducing.skip = "sendFileDescriptor producer was not paused"'>> src/twisted/internet/test/test_unix.py
+    echo 'UNIXTestsBuilder_SelectReactorTests.test_sendFileDescriptorTriggersPauseProducing.skip = "sendFileDescriptor producer was not paused"'>> src/twisted/internet/test/test_unix.py
+
+    echo 'FileObserverTests.test_getTimezoneOffsetEastOfUTC.skip = "mktime argument out of range"'>> src/twisted/test/test_log.py
+    echo 'FileObserverTests.test_getTimezoneOffsetWestOfUTC.skip = "mktime argument out of range"'>> src/twisted/test/test_log.py
+    echo 'FileObserverTests.test_getTimezoneOffsetWithoutDaylightSavingTime.skip = "tuple differs, values not"'>> src/twisted/test/test_log.py
 
+    echo 'MulticastTests.test_joinLeave.skip = "No such device"'>> src/twisted/test/test_udp.py
+    echo 'MulticastTests.test_loopback.skip = "No such device"'>> src/twisted/test/test_udp.py
+    echo 'MulticastTests.test_multicast.skip = "Reactor was unclean"'>> src/twisted/test/test_udp.py
+    echo 'MulticastTests.test_multiListen.skip = "No such device"'>> src/twisted/test/test_udp.py
+
+    echo 'DomishExpatStreamTests.test_namespaceWithWhitespace.skip = "syntax error: line 1, column 0"'>> src/twisted/words/test/test_domish.py
+
+    # not packaged
+    substituteInPlace src/twisted/test/test_failure.py \
+      --replace "from cython_test_exception_raiser import raiser  # type: ignore[import]" "raiser = None"
+  ''
   # Patch t.p._inotify to point to libc. Without this,
   # twisted.python.runtime.platform.supportsINotify() == False
-  postPatch = lib.optionalString stdenv.isLinux ''
+  + lib.optionalString stdenv.isLinux ''
     substituteInPlace src/twisted/python/_inotify.py --replace \
       "ctypes.util.find_library(\"c\")" "'${stdenv.glibc.out}/lib/libc.so.6'"
   '';
 
-  # Generate Twisted's plug-in cache.  Twisted users must do it as well.  See
+  # Generate Twisted's plug-in cache. Twisted users must do it as well. See
   # http://twistedmatrix.com/documents/current/core/howto/plugin.html#auto3
-  # and http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=477103 for
-  # details.
+  # and http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=477103 for details.
   postFixup = ''
     $out/bin/twistd --help > /dev/null
   '';
 
+  checkInputs = [
+    git
+    glibcLocales
+    pyhamcrest
+  ] ++ passthru.extras-require.conch
+  ++ passthru.extras-require.tls;
+
   checkPhase = ''
-    ${python.interpreter} -m unittest discover -s src/twisted/test
+    export SOURCE_DATE_EPOCH=315532800
+    export PATH=$out/bin:$PATH
+    ${python.interpreter} -m twisted.trial -j$NIX_BUILD_CORES twisted
   '';
-  # Tests require network
-  doCheck = false;
+
+  passthru.tests = {
```

Comment:
What's all this?

Reply:
@exarkun see #168407 for an explanation.

It's still a slightly experimental approach to testing heavily depended packages but if we find it causes problems we can always abandon the idea.